src/com/scottyplunkett/server/Router.java

```diff
@@ -4,17 +4,23 @@
 import java.nio.file.Paths;
 
 public class Router {
-        public static Path route(String requestLine) {
-            if ("GET / HTTP/1.1".equals(requestLine))
-                return Paths.get("pages/helloworld.html");
-            else if ("GET /nicole HTTP/1.1".equals(requestLine))
-                return Paths.get("pages/nicole.html");
-            else if ("GET /paul HTTP/1.1".equals(requestLine))
-                return Paths.get("pages/paul.html");
-            else if ("GET /josh HTTP/1.1".equals(requestLine))
-                return Paths.get("pages/josh.html");
-            else if (requestLine.split("\\?").length <= 1) {
-                return Paths.get("pages/404.html");
-                } else return Paths.get("pages/dynamic.html");
+    public static Path route(String requestLine) {
+        return hasParameters(requestLine) ? Paths.get("pages/dynamic.html") : walkRoutes(requestLine);
+    }
+
+    private static Path walkRoutes(String requestLine) {
+        switch (requestLine) {
+            case "GET / HTTP/1.1" : return Paths.get("pages/helloworld.html");
+            case "GET /nicole HTTP/1.1" : return Paths.get("pages/nicole.html");
+            case "GET /paul HTTP/1.1" : return Paths.get("pages/paul.html");
+            case "GET /josh HTTP/1.1" : return Paths.get("pages/josh.html");
+            case "GET /file1 HTTP/1.1" : return Paths.get("pages/file1");
+            case "GET /coffee HTTP/1.1" : return Paths.get("pages/418.html");
+            default : return Paths.get("pages/404.html");
         }
+    }
+
+    private static boolean hasParameters(String requestLine) {
+        return requestLine.split("\\?").length > 1 ? true : false;
```

Comment:
Do you need to explicitly return true and false here?

Reply:
Ha, good catch! I don't see why I would.